backend/db/utils.py

```diff
@@ -130,21 +182,107 @@ def sql_query(
         raise RuntimeError(f'Got an error [{err}] executing the following statement:\n{query}, {json.dumps(params, indent=2)}')
 
 
-def get_obj_by_id(con, table: str, pkey: str, obj_id: Union[str, int]):
+def delete_obj_by_composite_key(con, table: str, key_ids: Dict[str, Union[str, int]]):
     """Get object by ID"""
-    return sql_query(con, f'SELECT * FROM {table} WHERE {pkey} = (:obj_id)', {'obj_id': obj_id}, return_with_keys=False)
+    keys_str = ' AND '.join([f'{key} = (:{key})' for key in key_ids.keys()])
+    return run_sql(
+        con, f'DELETE FROM {table} WHERE {keys_str}',
+        {f'{key}': _id for key, _id in key_ids.items()})
 
-def insert_from_dict(con: Connection, table: str, d: Dict, skip_if_already_exists=True):
-    """Insert row into dictionary from a dictionary"""
-    # TODO: check whether already exists?
-    #   doing this already in fetch_object_and_add_to_db
-    #   like: already_in_db = get_obj_by_id(con, table, obj_pkey(table), object_id)
-    #   but other functions that call this should probably check or have this function check
 
+def get_obj_by_composite_key(con, table: str, keys: List[str], obj: Dict) -> List[Dict]:
```

Comment:
FYI: several helper functions:
- get_obj_by_composite_key
- get_objs_by_composite_key
- get_obj_by_id
- get_objs_by_id

I was considering combining so that there are only 2 funcs that can accept 1 key, or a list of keys.